### PR TITLE
Feature/script supersede lower versions

### PR DESCRIPTION
**Pull request by filipw**
Conflicting assembly versions in scripting are typically resolved via [identity comparer](https://github.com/OmniSharp/omnisharp-roslyn/blob/master/src/OmniSharp.Script/ScriptHelper.cs#L63), which we already use. 

However, scripting scenarios in Roslyn additionally set [a special flag in the ReferenceManager](https://github.com/dotnet/roslyn/blob/version-2.6.0-beta3/src/Compilers/Core/Portable/ReferenceManager/CommonReferenceManager.State.cs#L473-L484) to enable additional tie-breaking behavior in case of conflicting references. This flag [is propagated from](https://github.com/dotnet/roslyn/blob/version-2.6.0-beta3/src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs#L199) the compilation options, and is always implicitly injected for script compilations.

To mirror 1:1 the runtime behavior of scripts, we should ensure the same flag is set at tooling level (at the moment we have it set to to `false` which can lead to edge case false-positive intellisense errors). Unfortunately the flag is internal so it's set with reflection (along with a few other things we do with reflection in script tooling ☹️).

### CODE REVIEW

File: src/OmniSharp.Script/ScriptHelper.cs [hunk 1]
```diff
@@ -73,6 +74,14 @@ private CSharpCompilationOptions CreateCompilationOptions()
                 topLevelBinderFlagsProperty?.SetValue(compilationOptions, ignoreCorLibraryDuplicatedTypesValue);
             }
 
+            // in scripts, the option to supersede lower versions is ALWAYS enabled
+            // see: https://github.com/dotnet/roslyn/blob/version-2.6.0-beta3/src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs#L199
+            var referencesSupersedeLowerVersionsProperty = typeof(CompilationOptions).GetProperty(ReferencesSupersedeLowerVersionsProperty, BindingFlags.Instance | BindingFlags.NonPublic);
+            if (referencesSupersedeLowerVersionsProperty != null)
+            {
+                referencesSupersedeLowerVersionsProperty?.SetValue(compilationOptions, true);
```
Comment (DustinCampbell) [inline src/OmniSharp.Script/ScriptHelper.cs]: Is the `?.` needed after checking for null above?

Comment (david-driscoll) [reply src/OmniSharp.Script/ScriptHelper.cs]: Technically the if could be removed couldn't it?

Comment (filipw) [reply src/OmniSharp.Script/ScriptHelper.cs]: you are both right 😀

**Comment by filipw** [comment]
I will merge this now 😀

screen shot 2018-02-13 at 14 40 40

**Pull request merged**